client/components/marketing-survey/cancel-purchase/index.jsx

```diff
@@ -0,0 +1,357 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import shuffle from 'lodash/shuffle';
+
+/**
+ * Internal Dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLegend from 'components/forms/form-legend';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextarea from 'components/forms/form-textarea';
+
+const CancelPurchaseSurvey = React.createClass( {
+	propTypes: {
+		surveyStep: React.PropTypes.number.isRequired,
+		showSurvey: React.PropTypes.bool.isRequired,
+		defaultContent: React.PropTypes.node.isRequired,
+		callbackParent: React.PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		// shuffle reason order, but keep anotherReasonOne last
+		const questionOneOrder = shuffle( [
+			'couldNotInstall',
+			'tooHard',
+			'didNotInclude',
+			'onlyNeedFree'
+		] );
+		questionOneOrder.push( 'anotherReasonOne' );
+
+		const questionTwoOrder = shuffle( [
+			'stayingHere',
+			'otherWordPress',
+			'differentService',
+			'noNeed'
+		] );
+		questionTwoOrder.push( 'anotherReasonTwo' );
+
+		return {
+			questionOneRadio: null,
+			questionOneText: '',
+			questionOneOrder: questionOneOrder,
+			questionTwoRadio: null,
+			questionTwoText: '',
+			questionTwoOrder: questionTwoOrder,
+			questionThreeText: ''
+		};
+	},
+
+	handleRadioOne( event ) {
+		const newState = this.state;
+		newState.questionOneRadio = event.currentTarget.value;
+		newState.questionOneText = '';
+		this.setState( newState );
+		this.props.callbackParent( newState );
```

Comment:
I think this function (and all the handle functions) could be a little more clear with a rename to indicate when they are triggered and a tweak to how we assign the newState.
```js
onRadioOneChange( event )  {
  const newState = {
    ...this.state, 
    questionOneRadio: event.currentTarget.value,
    questoneOneText: '',
  };
  this.setState( newState ):
  this.props.onInputChange( newState );
}
```